pax/plugins/peak_processing/PeakAreaCorrections.py

```diff
@@ -64,3 +69,223 @@ def transform_event(self, event):
+
+                # select pulses inside this peak window
+                pulses_pd=pulses_pd[(pulses_pd["p_left"] <= end_sample) & (pulses_pd["p_right"] >= start_sample)]
+                # Start constructing the WFs
+                for pmtid in range(0, nPmts):
+                    channel_sat_left[pmtid]=end_sample
+                    channel_sat_right[pmtid]=start_sample
+                    index_all=pulses_pd[(pulses_pd.p_channel==pmtid)].p_id.values # all pulses in this WF
+                    adc_conversion=dsputils.adc_to_pe(self.config, channel=pmtid)
+                    
+                    for index in index_all:
+                        p=pulses[int(index)]
+                        p_left=p.left
+                        p_right=p.right
+                        raw_data_adc = p.raw_data.astype(np.float16)
+                        for sample in range(start_sample,end_sample):
+                            if(p_left<=sample and p_right>=sample):
+                                adc=raw_data_adc[sample-p.left]
+                                adc=16000.0-adc 
+                                wf_raw[pmtid,sample-start_sample]=adc*adc_conversion
+                            
+                                # determine whether there is saturation
+                                if (adc>=16000.0-0.5):
+                                    channel_sat_status[pmtid]=1
+                                    channel_sat_samples[pmtid]+=1
+                                    channel_sat_left[pmtid]=np.minimum(sample, channel_sat_left[pmtid])
+                                    channel_sat_right[pmtid]=np.maximum(sample, channel_sat_right[pmtid])
+                                    
+                # sum of WFs
+                wf_sum_all=np.sum(wf_raw,axis=0) # all PMTs including the saturated channels
+                # first deal with non-saturated PMTS, define Sum of nan_saturated WF 
+                wf_sum_nan_saturate=np.zeros_like(wf_sum_all) # Sum of non-saturated 
+                wf_sum_saturate=np.zeros_like(wf_sum_all) # Sum of non-saturated
+                for sample in range(start_sample,end_sample):
+                    for pmtid in range(0, nPmts):
+                        # first deal with non-saturated PMTS
+                        if(channel_sat_samples[pmtid]<1):
+                            wf_sum_nan_saturate[sample-start_sample]+=wf_raw[pmtid,sample-start_sample]
+                        else:
+                            wf_sum_saturate[sample-start_sample]+=wf_raw[pmtid,sample-start_sample]
+                area_nan_saturate=np.sum(wf_sum_nan_saturate) # Area of Total non-saturated WFs 
+                # The apply correction on saturated samples, need the shape of non-saturated WFs
+                # re-define pulse shape based on the derived model data
+                max_sample=np.argmax(wf_sum_nan_saturate) #sample sample at maximum
```

Comment:
__[flake8]__
* [E225] missing whitespace around operator
* [E261] at least two spaces before inline comment
* [E262] inline comment should start with '# '

<!-- Comment by [SideCI](https://sideci.com) -->